src/sql/workbench/services/objectExplorer/browser/asyncServerTreeDataSource.ts

```diff
@@ -45,25 +47,19 @@ export class AsyncServerTreeDataSource implements IAsyncDataSource<ConnectionPro
 	public async getChildren(element: ServerTreeElement): Promise<ServerTreeElement[]> {
 		try {
 			if (element instanceof ConnectionProfile) {
-				return await TreeUpdateUtils.getAsyncConnectionNodeChildren(element, this._connectionManagementService, this._objectExplorerService);
+				return await TreeUpdateUtils.getAsyncConnectionNodeChildren(element, this._connectionManagementService, this._objectExplorerService, this._configurationService);
 			} else if (element instanceof ConnectionProfileGroup) {
-				return (element as ConnectionProfileGroup).getChildren();
+				return element.getChildren();
 			} else if (element instanceof TreeNode) {
 				if (element.children) {
 					return element.children;
 				} else {
 					return await this._objectExplorerService.resolveTreeNodeChildren(element.getSession()!, element);
 				}
 			}
-		} catch (err) {
-			if (element instanceof TreeNode) {
-				element.errorStateMessage = err.message ?? err;
-			}
-			if (err.message) {
-				this.showError(err.message);
-			}
-
-			throw err;
+		} catch (error) {
+			this.showError(error);
```

Comment:
This doesn't seem related to your description of the changes - why did you remove the TreeNode check?